src/esp/physics/bullet/BulletPhysicsManager.cpp

```diff
@@ -261,24 +270,121 @@ RaycastResults BulletPhysicsManager::castRay(const esp::geo::Ray& ray,
   return results;
 }
 
-int BulletPhysicsManager::getNumActiveContactPoints() {
-  int pointCount = 0;
+void BulletPhysicsManager::lookUpObjectIdAndLinkId(
+    const btCollisionObject* colObj,
+    int* objectId,
+    int* linkId) const {
+  ASSERT(objectId);
+  ASSERT(linkId);
+
+  *linkId = -1;
+  // If the lookup fails, default to the stage. TODO: better error-handling.
+  *objectId = -1;
+
+  if (collisionObjToObjIds_->count(colObj)) {
+    int rawObjectId = collisionObjToObjIds_->at(colObj);
+    if (existingObjects_.count(rawObjectId)) {
+      *objectId = rawObjectId;
+      return;
+    }
+    // TODO: will come with AO support
+    /* else if (existingArticulatedObjects_.count(rawObjectId)) {
```

Comment:
What's going on here? This function looks half-finished.

Reply:
Just trying to show why linkId is part of this function at all, since it is still valid for rigids. This block should be uncommented with AOs.